### PR TITLE
Added MacOS with M1 build

### PR DESCRIPTION
**Pull request by asenyaev**
1. Created MacOS with M1 build using arm64.
2. Added exceptions for the workflows, which will not trigger other builds if there are no changes in these files.
3. Added exceptions for `arm64` platform in `pyproject.toml`

### CODE REVIEW

File: .github/workflows/build_wheels_macos_m1.yml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        submodules: false
+        fetch-depth: 0
+
+    - name: Update submodules
+      if: github.event_name == 'pull_request'
+      run: git submodule update --remote
+
+    - name: build script
+      run: |
+        git submodule update --init multibuild
```
Comment (sergregory) [inline .github/workflows/build_wheels_macos_m1.yml:48]: We do not need multibuild here, as we do all the build steps manually. To be honest, I've started thinking we do not need multibuild for all other builders too, as our required steps are simple and we already do them for OSX and windows. But this is a discussion for the other PR)

**Review by sergregory — APPROVED** [review]

File: .github/workflows/build_wheels_macos_m1.yml (hunk 1 above)
Comment (asenyaev) [reply .github/workflows/build_wheels_macos_m1.yml:48]: Okay, I will change the flow of the build without multibuild and prepare another PR.

**Comment by luisdemarchi** [comment]
@sergregory  It's been 14 hours since this merge was done, but it's been 8 hours since I needed to do the manual build. Where did I go wrong?

**Comment by sergregory** [comment]
@luisdemarchi probably, you did everything right - having a merge here doesn't mean we have a new release deployed to PyPI. We try to align the release cycle with the opencv release cycle, so now we're working on 4.5.3 and expect it to be ready this week.

**Comment by DanLipsitt** [comment]
@sergregory Any chance of deploying this first, since right now there's no release for M1 at all?

**Pull request merged**